### PR TITLE
fix(vendor/scoped): bump k8s version to 1.24, go version to 1.18 and fix scoped client

### LINKED ISSUE #2762: Bump repository to Go 1.18
## Feature Request

**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is. Example: "I have an issue when (...)"

Update the repository to Go 1.18.

**Describe the solution you'd like**
A clear and concise description of what you want to happen. Add any considered drawbacks.

AC:
- Update the root go.mod and bump the Go version to 1.18.
- Update any Makefile or GHA workflow references to Go 1.17.

### LINKED ISSUE #2792: Bump dependencies to support Kubernetes 1.24
## Feature Request

**Describe the solution you'd like**
OLM should support Kubernetes 1.24

I believe this would include upgrading the following dependencies:
- github.com/openshift/api --> https://github.com/openshift/api/commit/2c9b9549d9561f10c81c415c101e1fb11fd00cc6
- github.com/openshift/client-go --> https://github.com/openshift/client-go/commit/d1f246d64dd9f994a58b095e82cccf831cfe2d6a
- github.com/operator-framework/api --> v0.15.0
- github.com/operator-framework/operator-registry --> v1.23.0
- helm.sh/helm/v3 --> 3.9.0
- sigs.k8s.io/controller-runtime --> v0.12.1
- sigs.k8s.io/controller-tools --> v0.9.0
- k8s.io/* --> 0.24.0

### PR DESCRIPTION
**Pull request by dinhxuanvu**
Signed-off-by: Vu Dinh <vudinh@outlook.com>

**Description of the change:**

- Update k8s version to the latest 1.24
- Update go version to latest minor version 1.18
- Fix scoped client to handle token secret under new k8s 1.24 changes

**Motivation for the change:**

**Architectural changes:**

**Testing remarks:**

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage
- [ ] Sufficient end-to-end test coverage
- [ ] Bug fixes are accompanied by regression test(s)
- [ ] e2e tests and flake fixes are accompanied evidence of flake testing, e.g. executing the test 100(0) times
- [ ] tech debt/todo is accompanied by issue link(s) in comments in the surrounding code
- [ ] Tests are comprehensible, e.g. Ginkgo DSL is being used appropriately
- [ ] Docs updated or added to `/doc`
- [ ] Commit messages sensible and descriptive
- [ ] Tests marked as `[FLAKE]` are truly flaky and have an issue
- [ ] Code is properly formatted

Closes #2792
Closes #2762

### CODE REVIEW

**Review by timflannagan — COMMENTED** [review]
Bump the GHA workflow's go-version in the checkout action too?

**Comment by dinhxuanvu** [comment]
> Bump the GHA workflow's go-version in the checkout action too?

Will do

File: pkg/lib/scoped/token_retriever.go [hunk 1]
```diff
@@ -50,29 +49,45 @@ func (r *BearerTokenRetriever) Retrieve(reference *corev1.ObjectReference) (toke
...
+	secrets := filterSecretsBySAName(sa.Name, seList)
 
-			err = getErr
-			break
+	for _, ref := range sa.Secrets {
+		if _, ok := secrets[ref.Name]; !ok {
+			logger.Warnf("skipping secret %s: secret not found", ref.Name)
+			continue
 		}
+	}
```
Comment (timflannagan) [inline pkg/lib/scoped/token_retriever.go:64]: We're iterating over the list of SA secrets, purely for logging purposes, here? Is that right?

File: pkg/lib/scoped/token_retriever.go [hunk 2]
```diff
@@ -50,29 +49,45 @@ func (r *BearerTokenRetriever) Retrieve(reference *corev1.ObjectReference) (toke
...
 	return
 }
+
+// filterSecretsBySAName returna a maps of secrets that are associated with a
+// specific ServiceAccount via annotations kubernetes.io/service-account.name
+func filterSecretsBySAName(saName string, secrets *corev1.SecretList) map[string]*corev1.Secret {
+	secretMap := make(map[string]*corev1.Secret)
+	for _, ref := range secrets.Items {
+		annotations := ref.GetAnnotations()
+		value := annotations[corev1.ServiceAccountNameKey]
```
Comment (timflannagan) [inline pkg/lib/scoped/token_retriever.go:86]: Do we need to nil check the annotations map here before indexing for the SA name key, or do maps handle this instruction fine?

File: go.mod [hunk 3]
```diff
@@ -56,12 +56,26 @@ require (
...
+	go.opentelemetry.io/contrib => go.opentelemetry.io/contrib v0.20.0
+	go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp => go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.20.0
+	go.opentelemetry.io/otel/exporters/otlp => go.opentelemetry.io/otel/exporters/otlp v0.20.0
+	go.opentelemetry.io/otel/metric => go.opentelemetry.io/otel/metric v0.20.0
+	go.opentelemetry.io/otel/oteltest => go.opentelemetry.io/otel/oteltest v0.20.0
+	go.opentelemetry.io/otel/sdk/export/metric => go.opentelemetry.io/otel/sdk/export/metric v0.20.0
+	go.opentelemetry.io/otel/sdk/metric => go.opentelemetry.io/otel/sdk/metric v0.20.0
+	go.opentelemetry.io/otel/trace => go.opentelemetry.io/otel/trace v0.20.0
+	go.opentelemetry.io/proto/otlp => go.opentelemetry.io/proto/otlp v0.7.0
+)
```
Comment (timflannagan) [inline go.mod:71]: Note for our future selves: this was needed due to Kubernetes 1.24 requiring a new version of helm, which was problematic as we ran into module issues when bumping that dependency:

```
go: finding module for package go.opentelemetry.io/otel/internal/metric
go: finding module for package go.opentelemetry.io/otel/unit
go: finding module for package go.opentelemetry.io/otel/semconv
go: finding module for package rsc.io/letsencrypt
go: found rsc.io/letsencrypt in rsc.io/letsencrypt v0.0.3
go: found go.opentelemetry.io/otel/internal/metric in go.opentelemetry.io/otel/internal/metric v0.27.0
go: finding module for package go.opentelemetry.io/otel/metric/registry
go: finding module for package go.opentelemetry.io/otel/semconv
github.com/operator-framework/operator-lifecycle-manager/pkg/package-server/server imports
	k8s.io/apiserver/pkg/server/options imports
	go.opentelemetry.io/otel/semconv: module go.opentelemetry.io/otel@latest found (v1.7.0), but does not contain package go.opentelemetry.io/otel/semconv
github.com/operator-framework/operator-lifecycle-manager/pkg/package-server/server imports
	k8s.io/apiserver/pkg/server/options imports
	go.opentelemetry.io/otel/exporters/otlp/otlpgrpc imports
	go.opentelemetry.io/otel/exporters/otlp imports
	go.opentelemetry.io/otel/sdk/metric/controller/basic imports
	go.opentelemetry.io/otel/metric/registry: module go.opentelemetry.io/otel/metric@latest found (v0.30.0), but does not contain package go.opentelemetry.io/otel/metric/registry
```

It appears the issue here is that helm uses an incompatible containerd version which the opentelemetry module doesn't play well with. Aftering searching github for how others were tackling this through the filename:go.mod replace opentelemetry search query, and filtering by the newest matches, we found projects replace pinning the opentelemtry module version.

**Pull request merged**